runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/control/RemoteExecutionTest.java

```diff
@@ -630,72 +644,100 @@ public void process(ProcessContext ctxt) {
+                .setUrn(MonitoringInfoConstants.Urns.USER_SUM_INT64)
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAMESPACE, RemoteExecutionTest.class.getName())
+                .setLabel(MonitoringInfoConstants.Labels.NAME, MetricsDoFn.START_USER_COUNTER_NAME);
+            builder.setLabel(MonitoringInfoConstants.Labels.PTRANSFORM, testPTransformId);
+            builder.setInt64SumValue(10);
+            matchers.add(MonitoringInfoMatchers.matchSetFields(builder.build()));
 
-                  @Override
-                  public Coder<V> valueCoder() {
-                    return elementCoder.getValueCoder();
-                  }
+            builder = new SimpleMonitoringInfoBuilder();
+            builder
+                .setUrn(MonitoringInfoConstants.Urns.USER_SUM_INT64)
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAMESPACE, RemoteExecutionTest.class.getName())
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAME, MetricsDoFn.FINISH_USER_COUNTER_NAME);
+            builder.setLabel(MonitoringInfoConstants.Labels.PTRANSFORM, testPTransformId);
+            matchers.add(not(MonitoringInfoMatchers.matchSetFields(builder.build())));
 
-                  @Override
-                  public Iterable<V> get(K key, W window) {
-                    return (Iterable) sideInputData;
-                  }
-                };
-              }
-            });
+            // User Distributions.
+            builder
+                .setUrn(MonitoringInfoConstants.Urns.USER_DISTRIBUTION_INT64)
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAMESPACE, RemoteExecutionTest.class.getName())
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAME,
+                    MetricsDoFn.PROCESS_USER_DISTRIBUTION_NAME);
+            builder.setLabel(MonitoringInfoConstants.Labels.PTRANSFORM, testPTransformId);
+            builder.setInt64DistributionValue(DistributionData.create(1, 1, 1, 1));
+            matchers.add(MonitoringInfoMatchers.matchSetFields(builder.build()));
 
-    String testPTransformId = "create/ParMultiDo(Anonymous)";
-    BundleProgressHandler progressHandler =
-        new BundleProgressHandler() {
-          @Override
-          public void onProgress(ProcessBundleProgressResponse progress) {}
+            builder = new SimpleMonitoringInfoBuilder();
+            builder
+                .setUrn(MonitoringInfoConstants.Urns.USER_DISTRIBUTION_INT64)
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAMESPACE, RemoteExecutionTest.class.getName())
+                .setLabel(
+                    MonitoringInfoConstants.Labels.NAME, MetricsDoFn.START_USER_DISTRIBUTION_NAME);
+            builder.setLabel(MonitoringInfoConstants.Labels.PTRANSFORM, testPTransformId);
+            builder.setInt64DistributionValue(DistributionData.create(10, 1, 10, 10));
```

Comment:
Are these values likely to easily change? You could consider writing new matchers in MonitoringInfoMatchers to match that the values are non 0 or something instead to make it easier to maintain.

Or write matchers that just verify a few fields, unless you really want to verify everything is set on every MonitoringInfo. Might also make it simpler to maintain. Unless you think they are all relevant. Up to your discretion here.

Reply:
The values should only change if the test changes since they are expected to be emitted once. Only the msec counters are non-deterministic really.

Also, the MonitoringInfo matcher only compares set fields and ignores fields that aren't set.